src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -124,22 +125,32 @@ private static LinqEnumerableMapping BuildLinqMapping(MappingBuilderContext ctx,
         return new LinqEnumerableMapping(ctx.Source, ctx.Target, elementMapping, selectMethod, collectMethod);
     }
 
-    private static bool HasEnumerableConstructor(MappingBuilderContext ctx, ITypeSymbol typeSymbol)
+    private static INamedTypeSymbol? GetTypeConstructableFromEnumerable(MappingBuilderContext ctx, ITypeSymbol typeSymbol)
     {
         if (ctx.Target is not INamedTypeSymbol namedType)
-            return false;
+            return null;
 
         var typedEnumerable = ctx.Types.Get(typeof(IEnumerable<>)).Construct(typeSymbol);
-
-        return namedType.Constructors.Any(
+        var hasCtor = namedType.Constructors.Any(
             m => m.Parameters.Length == 1 && SymbolEqualityComparer.Default.Equals(m.Parameters[0].Type, typedEnumerable)
         );
+        if (hasCtor)
+            return namedType;
+
+        if (ctx.CollectionInfos!.Target.Type is CollectionType.ISet or CollectionType.IReadOnlySet)
+            return ctx.Types.Get(typeof(HashSet<>)).Construct(typeSymbol);
```

Comment:
This seems to be not covered by tests

Reply:
It is but only by the .NET Framework integration tests which don't run under coverage. For the others `.ToHashSet()` is used.